### PR TITLE
Add corr, covar_samp, and covar_pop Presto aggregate functions

### PR DESCRIPTION
**Pull request by mbasmanova**

### CODE REVIEW

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+};
+constexpr CovarIndices kCovarIndices{1, 2, 3, 0};
+
+// Indices into RowType representing intermediate results of corr. Columns
+// appear in alphabetical order.
+struct CorrIndices : public CovarIndices {
+  int32_t m2X;
+  int32_t m2Y;
+};
+constexpr CorrIndices kCorrIndices{{1, 4, 5, 0}, 2, 3};
```
Comment (kgpai) [inline velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:39]: Nit: Can you add a comment on why these have these values ?

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  });
+
+  createDuckDbTable({data});
+
+  auto aggName = GetParam();
+  testGlobalAgg(aggName, data);
+  testGroupBy(aggName, data);
+}
+
+TEST_P(CovarianceAggregationTest, someNulls) {
```
Comment (kgpai) [inline velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:91]: Is it possible to add a test with a nan in the values ?

**Review by kgpai — APPROVED** [review]
LGTM .

File: velox/functions/prestosql/aggregates/CovarianceAggregates.cpp [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+      const RowVector* rowVector,
+      const CovarIndices& indices = kCovarIndices)
+      : count_{asSimpleVector<int64_t>(rowVector, indices.count)},
+        meanX_{asSimpleVector<double>(rowVector, indices.meanX)},
+        meanY_{asSimpleVector<double>(rowVector, indices.meanY)},
+        c2_{asSimpleVector<double>(rowVector, indices.c2)} {}
+
+  void mergeInto(CovarAccumulator& accumulator, vector_size_t row) {
+    accumulator.merge(
+        count_->valueAt(row),
```
[the remark this answers is not in the document]
Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/CovarianceAggregates.cpp:132]: Yes. The values are produced by extractAccumulator which doesn't make nulls.

File: velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp (hunk 2 above)
Comment (mbasmanova) [reply velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:91]: It looks like DuckDB doesn't allow storing NaNs in a table. I'm getting "Out of Range Error: Invalid double value nan" error. @Mytherin Mark, is there a way?

```
terminate called after throwing an instance of 'duckdb::OutOfRangeException'
  what():  Out of Range Error: Invalid double value nan
*** Aborted at 1638561988 (Unix time, try 'date -d @1638561988') ***
*** Signal 6 (SIGABRT) (0x2aca000b812c) received by PID 753964 (pthread TID 0x7ff2af459d80) (linux TID 753964) (maybe from PID 753964, UID 10954) (code: -6), stack trace: ***
    @ 00000000000192fd folly::symbolizer::(anonymous namespace)::innerSignalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:449
    @ 00000000000102ab folly::symbolizer::(anonymous namespace)::signalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:470
    @ 0000000000000000 (unknown)
    @ 000000000003e530 __GI_raise
    @ 000000000002551c __GI_abort
    @ 00000000000a07d4 __gnu_cxx::__verbose_terminate_handler()
    @ 000000000009e4a5 __cxxabiv1::__terminate(void (*)())
    @ 000000000009e510 std::terminate()
    @ 000000000009e7db __cxa_throw
    @ 000000000101559d duckdb::Value::Value(double)
                       ./velox/external/duckdb/duckdb.cpp:31332
    @ 00000000002b2ab7 duckdb::Value facebook::velox::exec::test::(anonymous namespace)::duckValueAt<(facebook::velox::TypeKind)6>(std::shared_ptr<facebook::velox::BaseVector> const&, int)
                       ./velox/exec/tests/utils/QueryAssertions.cpp:54
    @ 000000000029bde9 facebook::velox::exec::test::DuckDbQueryRunner::createTable(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::shared_ptr<facebook::velox::RowVector>, std::allocator<std::shared_ptr<facebook::velox::RowVector> > > const&)::$_0::operator()() const
                       ./velox/exec/tests/utils/QueryAssertions.cpp:397
```

Comment (Mytherin) [reply velox/functions/prestosql/aggregates/tests/CovarianceAggregationTest.cpp:91]: DuckDB does not currently allow NaNs into the system, they are instead converted to `NULL` values when ingested or an error is thrown when they are generated from a function call.

**Pull request closed without merging**